TWTValidation/JSON Validator/Parser/TWTJSONSchemaParser.m

```diff
@@ -300,12 +307,25 @@ - (void)parseReferenceSchema:(NSDictionary *)schema intoNode:(TWTJSONSchemaRefer
     NSString *referencePath = schema[TWTJSONSchemaKeywordRef];
 
     [self failIfObject:referencePath isNotKindOfClass:[NSString class] allowsNil:NO];
+
+    referencePath = [referencePath stringByReplacingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+
     NSArray *pathComponents = [referencePath componentsSeparatedByString:@"/"];
-    if (![pathComponents.firstObject isEqualToString:@"#"]) {
-        [self failWithErrorCode:TWTJSONSchemaParserErrorCodeInvalidValue object:referencePath format:@"Expected reference path to begin with # character"];
+    if ([pathComponents.firstObject isEqualToString:@"#"]) {
+        referenceNode.referencePathComponents = pathComponents;
+    } else if (![self.remoteSchemaManager attemptToConfigureFilePath:referencePath onReferenceNode:referenceNode]) {
+        [self failWithErrorCode:TWTJSONSchemaParserErrorCodeInvalidValue object:referencePath format:@"Reference path does not refer to the current schema or a valid file path"];
     }
 
-    referenceNode.referencePathComponents = pathComponents;
+    // See JSON Pointer doc (section 3. Syntax) for special characters
+    NSMutableArray *decodedComponents = [[NSMutableArray alloc] initWithCapacity:referenceNode.referencePathComponents.count];
+    for (NSString *component in referenceNode.referencePathComponents) {
+        NSString *decodedComponent = [[component stringByReplacingOccurrencesOfString:TWTJSONPointerSlashEncoding withString:TWTJSONPointerSlashValue] stringByReplacingOccurrencesOfString:TWTJSONPointerTildaEncoding withString:TWTJSONPointerTildaValue];
```

Comment:
[Tilda](https://upload.wikimedia.org/wikipedia/commons/thumb/2/2e/Tilda_Swinton_at_the_Deauville_Film_Festival.jpg/440px-Tilda_Swinton_at_the_Deauville_Film_Festival.jpg) and [tilde](https://en.wikipedia.org/wiki/Tilde) :smile: